extensions/positron-reticulate/src/extension.ts

```diff
@@ -484,20 +484,45 @@ class ReticulateRuntimeSession implements positron.LanguageRuntimeSession {
 		// tied to the R session.
 		// We have to send a restart to the R session, and send a reticulate::repl_python()
 		// command to it.
-		this.pythonSession.shutdown(positron.RuntimeExitReason.Restart);
-		await this.rSession.restart();
-		const rSession = await getRSession();
-		rSession.execute(
-			'reticulate::repl_python()',
-			'start-reticulate',
-			positron.RuntimeCodeExecutionMode.Interactive,
-			positron.RuntimeErrorBehavior.Continue
+		const restart = await positron.window.showSimpleModalDialogPrompt(
+			'Restarting reticulate',
+			'This is will also restart the parent R session. Are you sure you want to continue?',
+			'Yes',
+			'No'
 		);
+
+		if (!restart) {
+			return;
```

Comment:
This needs to throw an exception or otherwise notify the front end that the restart was cancelled. In my testing if you say "No" to this prompt, the front end is stuck in the "restarting" state since it is waiting for the runtime to shut down. 

<img width="867" alt="image" src="https://github.com/user-attachments/assets/4dc1e257-ec6b-4366-9a84-9564acb5cc87">

Reply:
Oops, thanks for catching this!